cmd/opentelemetry-collector/cmd/all-in-one/main.go

```diff
@@ -0,0 +1,237 @@
+	if err != nil {
+		handleErr(err)
+	}
+	cmd := svc.Command()
+	jConfig.AddFlags(v,
+		cmd,
+		app.AddComponentFlags,
+		storageFlags,
+		queryApp.AddFlags,
+	)
+
+	// parse flags to propagate Jaeger config file flag value to viper
+	cmd.ParseFlags(os.Args)
+	err = jflags.TryLoadConfigFile(v)
+	if err != nil {
+		handleErr(fmt.Errorf("could not load Jaeger configuration file %w", err))
+	}
+
+	go func() {
+		err = svc.Start()
+		handleErr(err)
+	}()
+
+	configCreatedWait.Wait()
+	exp := getStorageExporter(storageType, exporters)
+	if err := startQuery(v, svc.GetLogger(), exp); err != nil {
+		svc.ReportFatalError(err)
+	}
+}
+
+func getStorageExporter(storageType string, exporters configmodels.Exporters) configmodels.Exporter {
+	return exporters[fmt.Sprintf("jaeger_%s", storageType)]
+}
+
+func startQuery(v *viper.Viper, logger *zap.Logger, exporter configmodels.Exporter) error {
+	storageFactory, err := getFactory(exporter, v, logger)
+	if err != nil {
+		return err
+	}
+	spanReader, err := storageFactory.CreateSpanReader()
+	if err != nil {
+		return err
+	}
+	dependencyReader, err := storageFactory.CreateDependencyReader()
+	if err != nil {
+		return err
+	}
+	queryOpts := new(queryApp.QueryOptions).InitFromViper(v, logger)
+	queryServiceOptions := queryOpts.BuildQueryServiceOptions(storageFactory, logger)
+	queryService := querysvc.NewQueryService(
+		spanReader,
+		dependencyReader,
+		*queryServiceOptions)
+
+	tracerCloser := initTracer(logger)
+	server := queryApp.NewServer(logger, queryService, queryOpts, opentracing.GlobalTracer())
+	if err := server.Start(); err != nil {
+		return err
+	}
+
+	// TODO subscribe to service's lifecycle (shutdown) event and then terminate
+	// https://github.com/open-telemetry/opentelemetry-collector/issues/1033
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt)
+	<-c
+	return tracerCloser.Close()
+}
+
+func getFactory(exporter configmodels.Exporter, v *viper.Viper, logger *zap.Logger) (storage.Factory, error) {
+	switch exporter.Name() {
+	case "jaeger_elasticsearch":
+		archiveOpts := esStorage.NewOptions("es-archive")
+		archiveOpts.InitFromViper(v)
+		primaryConfig := exporter.(*elasticsearch.Config)
+		f := esStorage.NewFactory()
+		f.InitFromOptions(*esStorage.NewOptionsFromConfig(primaryConfig.Primary.Configuration, archiveOpts.Primary.Configuration))
+		if err := f.Initialize(metrics.NullFactory, logger); err != nil {
+			return nil, err
+		}
+		return f, nil
+	case "jaeger_cassandra":
+		archiveOpts := cassStorage.NewOptions("cassandra-archive")
+		archiveOpts.InitFromViper(v)
+		primaryConfig := exporter.(*cassandra.Config)
+		f := cassStorage.NewFactory()
+		f.InitFromOptions(cassStorage.NewOptionsFromConfig(primaryConfig.Primary.Configuration, archiveOpts.Primary.Configuration))
+		if err := f.Initialize(metrics.NullFactory, logger); err != nil {
+			return nil, err
+		}
+		return f, nil
+	case "memory":
+		return memory.GetFactory(), nil
```

Comment:
Why not badger, grpc_plugin?

Reply:
badger hasn't been implemented yet. I will add the `grpc-plugin`.